pytorch_lightning/plugins/checkpoint/torch.py

```diff
@@ -0,0 +1,23 @@
+from pathlib import Path
+from typing import Any, Dict, Union
+
+import pytorch_lightning as pl
+from pytorch_lightning.plugins.checkpoint.checkpoint import CheckpointIOPlugin
+from pytorch_lightning.utilities import rank_zero_warn
+from pytorch_lightning.utilities.cloud_io import atomic_save
+from pytorch_lightning.utilities.cloud_io import load as pl_load
+
+
+class TorchCheckpointIOPlugin(CheckpointIOPlugin):
+    def save_checkpoint(self, checkpoint: Dict[str, Any], path: Union[str, Path]) -> None:
+        try:
+            # write the checkpoint dictionary on the file
+            atomic_save(checkpoint, path)
+        except AttributeError as err:
+            key = pl.LightningModule.CHECKPOINT_HYPER_PARAMS_KEY
+            checkpoint.pop(key, None)
+            rank_zero_warn(f"Warning, `{key}` dropped from checkpoint. An attribute is not picklable: {err}")
+            atomic_save(checkpoint, path)
+
+    def load_checkpoint_file(self, path: Union[str, Path]) -> Dict[str, Any]:
+        return pl_load(path, map_location=(lambda storage, loc: storage))
```

Comment:
```suggestion
    def load_checkpoint_file(self, path: Union[str, Path], map_location: Callable) -> Dict[str, Any]:
        return pl_load(path, map_location=map_location)
```

Reply:
I'm a bit hesistant to add this to the class signature as this isn't always the case when loading from various sources, however I may be wrong cc @ananthsub

Reply:
I agree with @SeanNaren , `map_location` is specific to using `torch.load` 

this could be an optional parameter on the Torch plugin with a default value to maintain compatibility with the main interface, but it might not be needed here

Reply:
Done!